public/app/features/dashboard-scene/edit-pane/shared.ts

```diff
@@ -145,4 +162,34 @@ export const dashboardEditActions = {
       undo,
     });
   },
+
+  addVariable({ source, addedObject }: AddVariableActionHelperProps) {
+    const varsBeforeAddition = [...source.state.variables];
+
+    dashboardEditActions.addElement({
+      source,
+      addedObject,
+      perform() {
+        source.setState({ variables: [...varsBeforeAddition, addedObject] });
+      },
+      undo() {
+        source.setState({ variables: [...varsBeforeAddition] });
+      },
+    });
+  },
+  removeVariable({ source, removedObject }: RemoveVariableActionHelperProps) {
+    const varsBeforeRemoval = [...source.state.variables];
+
+    dashboardEditActions.removeElement({
+      source,
+      removedObject,
+      perform() {
+        removedObject.publishEvent(new ObjectRemovedFromCanvasEvent(removedObject), true);
```

Comment:
I don't think we need to emit this event. dashboardEditActions.removeElements emits an event that will result in the same outcome, clearing the selection.